pdns/syncres.cc

```diff
@@ -3508,8 +3527,10 @@ int SyncRes::doResolveAt(NsSet &nameservers, DNSName auth, bool flawedNSSet, con
             }
           }
           LOG(endl);
-          if (hitPolicy) //implies d_wantsRPZ
-            return -2;
+          if (hitPolicy) { //implies d_wantsRPZ
+            /* RPZ hit */
+            throw PolicyHitException();
+          }
         }
```

Comment:
There are still a few places in the code (and comments) were return -2 is used for a RPZ hit. Since `doResolve()` now  throws an exception for all these cases, these lines of code and commens should be adapted. Specifically: 767, 889, 3428 (hope I got them all)

Reply:
Done, thanks!

Reply:
Done, thanks!